istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,361 @@
+		Args: func(cmd *cobra.Command, args []string) error {
+			if name == "" {
+				return fmt.Errorf("expecting a service name")
+			}
+			if namespace == "" {
+				return fmt.Errorf("expecting a service namespace")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			u := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": collections.IstioNetworkingV1Alpha3Workloadgroups.Resource().APIVersion(),
+					"kind":       collections.IstioNetworkingV1Alpha3Workloadgroups.Resource().Kind(),
+					"metadata": map[string]interface{}{
+						"name":      name,
+						"namespace": namespace,
+					},
+				},
+			}
+			spec := &networkingv1alpha3.WorkloadGroup{
+				Template: &networkingv1alpha3.WorkloadEntryTemplate{
+					Metadata: &metav1.ObjectMeta{
+						Labels: convertToStringMap(labels),
+					},
+					Spec: &networkingv1alpha3.WorkloadEntry{
+						Ports:          convertToUnsignedInt32Map(ports),
+						ServiceAccount: serviceAccount,
+					},
+				},
+			}
+			wgYAML, err := generateWorkloadGroupYAML(u, spec)
+			if err != nil {
+				return err
+			}
+			_, err = cmd.OutOrStdout().Write(wgYAML)
+			return err
+		},
+	}
+	createGroupCmd.PersistentFlags().StringVar(&name, "name", "", "The name of the workload group")
+	createGroupCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "", "The namespace that the workload instances will belong to")
+	createGroupCmd.PersistentFlags().StringSliceVarP(&labels, "labels", "l", nil, "The labels to apply to the workload instances; e.g. -l env=prod,vers=2")
+	createGroupCmd.PersistentFlags().StringSliceVarP(&ports, "ports", "p", nil, "The incoming ports that the workload instances will expose")
+	createGroupCmd.PersistentFlags().StringVarP(&serviceAccount, "serviceAccount", "s", "default", "The service identity to associate with the workload instances")
+	return createGroupCmd
+}
+
+func generateWorkloadGroupYAML(u *unstructured.Unstructured, spec *networkingv1alpha3.WorkloadGroup) ([]byte, error) {
+	iSpec, err := unstructureIstioType(spec)
+	if err != nil {
+		return nil, err
+	}
+	u.Object["spec"] = iSpec
+	// remove creationTimestamp left behind by ObjectMeta
+	delete(iSpec["template"].(map[string]interface{})["metadata"].(map[string]interface{}), "creationTimestamp")
+
+	wgYAML, err := yaml.Marshal(u.Object)
+	if err != nil {
+		return nil, err
+	}
+	return wgYAML, nil
+}
+
+// Cluster inference from the current kubectl context only works for GKE
+func generateConfigCommand() *cobra.Command {
+	generateConfigCmd := &cobra.Command{
+		Use:   "generate-config",
+		Short: "Generates and packs all the required configuration files for deployment",
+		Long: `Takes in WorkloadGroup artifact, then generates and packs all the required configuration files for deployment. 
+This includes a MeshConfig resource, the cluster.env file, and necessary certificates and security tokens. 
```

Comment:
do we need certificates? I thought Istio was using tokens in 1.8? If Istio is not using tokens in 1.8, then remove the word "tokens".

Can you please provide documentation on token setup so the istio.io documentation can be updated?